source/slides/02_linux_basics.rst

```diff
@@ -0,0 +1,506 @@
+
+.. _Emacs Manual: https://www.gnu.org/software/emacs/manual/
+.. _Emacs Wiki: http://emacswiki.org/
+
+Editor questions?
+-----------------
+
+* Open an editor, find a cheat sheet, try to add some text
+* Modify the text: "``disemvowel``" it
+
+.. code-block:: bash
+
+    $ vim testvim.txt            $ emacs testemacs.txt
+    <i>                          Hello world!
+    Hello world!                 <alt + x>
+    <esc>                        replace-regexp
+    :%s/[aeiou]//g               <enter>
+    <enter>                      [aeiou]
+    :wq                          <enter>
+    <enter>                      <ctrl + x> <ctrl + s>
+                                 <ctrl + x> <ctrl + c>
+
+Git
+===
+
+Git
+---
+
+.. figure:: ../_static/Linus_Torvalds.jpeg
+    :align: left
+    :width: 90%
+
+git, noun. Brit.informal.
+1. an unpleasant or contemptible person.
+
+Setting up Git
+--------------
+
+* In VM:
+
+.. code-block:: bash
+
+    $ sudo yum install git
+    $ git config --global user.name "My Name"
+    $ git config --global user.email "myself@gmail.com"
+    $ git config --global core.editor "emacs"
+    $ git config --global push.default "current" # Or "nothing"
+
+Using Git Locally
+-----------------
+
+``$ git init``
+
+``$ git add <filename>``
+
+``$ git commit -m "I did a thing!"``
+
+* Undo things?
+  the `git book <http://git-scm.com/book/en/Git-Basics-Undoing-Things>`_ explains
+  well
+
+* Did I remember to commit that?::
+    
+    $ git status
+
+* What commits have I made lately?::
+
+    $ git log
+
+
+.. note::
+  This initializes a git repo. Use `man git-init` for more info.
+
+  This puts <filename> into the staging area. It isn't committed yet.  Use ``git
+  diff`` to see what changes aren't yet in staging.
+
+  This actually makes the commit. Use ``git status`` to see what's in staging
+  but not yet committed. Use ``git show`` or ``git log`` to see recent commits.
+
+What Not To Do
+--------------
+
+* Don't delete the .git directory
+* Avoid redundant copies of the same work in one revision
+* Don't make "oops, undoing that" commits.
+
+    * Use git commit --amend or git revert
+
+* Don't wait too long between commits
+
+    * You can squash them all together later
+
+* Don't commit secrets...
+
+* If you do commit secrets, `reset them`_
+
+.. _reset them: https://jordan-wright.github.io/blog/2014/12/30/why-deleting-sensitive-information-from-github-doesnt-save-you/
+
+Daily workflow
+--------------
+
+.. figure:: ../_static/gitflow.png
+    :scale: 65%
+    :align: right
+
+Pull -> Work -> Add changes -> Commit -> Push
+
+Larger projects have more complex workflows
+
+.. note:: The picture is of the Git Flow branching model, and you'll probably
+    see it every single time anyone explains Git branching and merging to you.
+
+Openstack
+=========
+
+Ground Rules
+------------
+
+* Do not attempt to login to other machines without
+  explicit permission
+* Do not do anything illegal (no torrenting)
```

Comment:
Might be better to just mention that you should follow the OSU AUP: http://fa.oregonstate.edu/gen-manual/acceptable-use-university-computing-resources. That should cover most of what's listed here. Although a short reminder is good too. I'm going to be basically copying this on the doc page too.